### PR TITLE
Some suggestions to product tour

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## Explanation
Some suggestions that contribute to https://github.com/MetaMask/metamask-extension/pull/18571/ that reduces some html nesting and utilizes media queries over `popupStatus`

I've created another PR to make these changes a lot less https://github.com/MetaMask/metamask-extension/pull/18678

### Before

https://user-images.githubusercontent.com/8112138/233269260-d286c934-3ec3-439b-8680-028fb308698a.mov

### After

https://user-images.githubusercontent.com/8112138/233269347-8dfcb9d3-506b-4251-a5ec-dd815a05b31d.mov

### CODE REVIEW

File: ui/components/multichain/product-tour-popover/product-tour-popover.js [hunk 1]
```diff
@@ -51,78 +50,75 @@ export const ProductTour = ({
       anchorElement={anchorElement}
       onHide={closeMenu}
       data-testid="multichain-product-tour-menu-popover"
+      {...props}
```
Comment (georgewrmarshall) [inline ui/components/multichain/product-tour-popover/product-tour-popover.js:53]: Double check that classes are added to menu component

File: ui/components/multichain/product-tour-popover/product-tour-popover.js [hunk 2]
```diff
@@ -51,78 +50,75 @@ export const ProductTour = ({
...
-              textAlign={TextAlign.Center}
-              variant={TextVariant.headingSm}
-              width={BLOCK_SIZES.FULL}
-              color={TextColor.infoInverse}
-            >
-              {title}
-            </Text>
-          </Box>
-
+        <Box
```
Comment (georgewrmarshall) [inline ui/components/multichain/product-tour-popover/product-tour-popover.js:69]: reduces some nesting

File: ui/components/multichain/product-tour-popover/product-tour-popover.scss [hunk 3]
```diff
@@ -34,6 +34,16 @@
...
+  &__header {
+    position: relative;
+  }
+
+  &__previous-icon {
+    position: absolute;
+    left: 0;
+    top: 0;
+  }
+
```
Comment (georgewrmarshall) [inline ui/components/multichain/product-tour-popover/product-tour-popover.scss:46]: Centers text when previous icon button is present unfortunatley we don't have position utility props for the Box component so need to add them as styles

File: ui/components/multichain/product-tour-popover/product-tour-popover.stories.js [hunk 4]
```diff
@@ -24,7 +24,13 @@ export default {
...
     },
     onClick: {
-      control: 'onClick',
+      action: 'onClick',
+    },
+    onHide: {
+      action: 'onHide',
+    },
+    closeMenu: {
+      action: 'closeMenu',
```
Comment (georgewrmarshall) [inline ui/components/multichain/product-tour-popover/product-tour-popover.stories.js:33]: Fixes some propType errors in the story

Screenshot 2023-04-20 at 2 42 15 PM

File: ui/components/component-library/avatar-base/avatar-base.js [hunk 5]
```diff
@@ -18,44 +18,51 @@ import { Text } from '../text';
...
-  size = AVATAR_BASE_SIZES.MD,
-  children,
-  backgroundColor = BackgroundColor.backgroundAlternative,
-  borderColor = BorderColor.borderDefault,
-  color = TextColor.textDefault,
-  className,
-  ...props
-}) => {
-  let fallbackTextVariant;
+export const AvatarBase = React.forwardRef(
```
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-base/avatar-base.js:21]: Needed for to pass refs down to the html element

File: ui/components/component-library/avatar-network/avatar-network.js [hunk 6]
```diff
@@ -14,70 +14,76 @@ import {
...
-  src,
-  showHalo,
-  color = TextColor.textDefault,
-  backgroundColor = BackgroundColor.backgroundAlternative,
-  borderColor = BorderColor.transparent,
-  className,
-  ...props
-}) => {
-  const [showFallback, setShowFallback] = useState(false);
+export const AvatarNetwork = React.forwardRef(
```
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-network/avatar-network.js:17]: Needed for to pass refs down to the html element

**Comment by NidhiKJha** [comment]
Thanks a lot George for the amazing suggestions. I am merging this PR

**Pull request merged**